core/src/Dscp/Crypto/Util.hs

```diff
@@ -0,0 +1,18 @@
+-- | Small utility functions
+
+module Dscp.Crypto.Util
+       ( constTimeEq
+       ) where
+
+import qualified Crypto.Util as CU (constTimeEq)
+import Data.ByteArray (ByteArrayAccess, convert)
+
+-- | Checks two bytestrings for equality without breaches for
+-- timing attacks.
+--
+-- Adapts 'Crypto.Util.constTimeEq' for any 'ByteArrayAccess' instance.
+-- See <http://codahale.com/a-lesson-in-timing-attacks/> for more info.
+constTimeEq
+    :: (ByteArrayAccess b1, ByteArrayAccess b2)
+    => b1 -> b2 -> Bool
+constTimeEq b1 b2 = CU.constTimeEq (convert b1) (convert b2)
```

Comment:
Lool, I wonder how many places are missing this function? For instance, our hashes are compared on a simple `==` inside.

Reply:
It's not that we need to use this everywhere - using it in inner logic of block validation, for example, makes no sense. We might just define `Eq` instances for all data which might be sensitive using this function, but then we might experience some detectable performance overhead without a good reason. Unfortunately, we need to _think_ when we compare some data with some another in order to decide whether or not to use this function.